src/keymap-manager.coffee

```diff
@@ -622,10 +629,16 @@ class KeymapManager
       # event. This means the current event has removed any hope that the queued
       # key events will ever match any binding. So we will clear the state and
       # start over after replaying the events in `terminatePendingState`.
-      @terminatePendingState()
+      @terminatePendingState() if event.type is 'keyup'
```

Comment:
What's the reasoning behind this change? I did some testing with my recent changes and it seems to work fine but I couldn't figure out why you added this.